js/dataframe/shared/Table.svelte

```diff
@@ -144,34 +144,42 @@
 	}
 
 	let _headers = make_headers(headers);
-	let old_headers: string[] | undefined;
+	let old_headers: string[] = headers;
 
 	$: {
 		if (!dequal(headers, old_headers)) {
-			trigger_headers();
+			_headers = make_headers(headers);
+			old_headers = JSON.parse(JSON.stringify(headers));
 		}
 	}
 
-	function trigger_headers(): void {
-		_headers = make_headers(headers);
-
-		old_headers = headers.slice();
-		trigger_change();
-	}
+	let data: { id: string; value: string | number }[][] = [[]];
+	let old_val: undefined | (string | number)[][] = undefined;
 
 	$: if (!dequal(values, old_val)) {
 		data = process_data(values as (string | number)[][]);
 		old_val = JSON.parse(JSON.stringify(values)) as (string | number)[][];
 	}
 
-	let data: { id: string; value: string | number }[][] = [[]];
-
-	let old_val: undefined | (string | number)[][] = undefined;
+	let previous_data_string = "[]";
+	let previous_headers_string = "[]";
```

Comment:
why arent these
```js
	let previous_data_string = JSON.stringify(
			data.map((row) => row.map((cell) => String(cell.value)));
	let previous_headers_string = JSON.stringify(...)
```

Won't this cause a change dispatch immediately because of the
```
 if (
			current_data_string !== previous_data_string ||
			current_headers_string !== previous_headers_string
		)
```
later

Reply:
It doesn't dispatch immediately, but I'm not sure why. I can make the change though

Reply:
Fixed

Reply:
initial dispatches often don't, but I think it's flaky. Better to remove